validate/src/command.rs

```diff
@@ -169,11 +175,11 @@ pub enum CommandValidationErrorType {
     /// Command description is invalid.
     DescriptionInvalid,
     /// Command name is invalid.
-    NameInvalid,
+    NameInvalid(NameValidationError),
```

Comment:
we shouldn't create a new error type here. just add variants to this enum for `ChatInputNameInvalid` or `MessageUserNameInvalid`.

Reply:
A name can be invalid for different reasons, how do you want to handle that without creating another error type?

Reply:
I don't mind adding more variants to the main error type, but having nested error types complicates the design too much IMO.